code/game/objects/items/tanks/tanks.dm

```diff
@@ -228,59 +219,71 @@
 
 	return remove_air(moles_needed)
 
-/obj/item/tank/process()
-	//Allow for reactions
-	air_contents.react()
-	check_status()
-
-/obj/item/tank/proc/check_status()
-	//Handle exploding, leaking, and rupturing of the tank
+/obj/item/tank/process(delta_time)
+	if(!air_contents)
+		return
 
+	//Allow for reactions
+	air_contents.react(src)
+	if(leaking)
+		var/turf/location = get_turf(src)
+		if(location)
+			var/datum/gas_mixture/leaked_gas = air_contents.remove_ratio(0.25)
+			location.assume_air(leaked_gas)
+			location.air_update_turf(FALSE, FALSE)
+
+	handle_tolerances(delta_time)
+
+/**
+ * Handles the minimum and maximum pressure tolerances of the tank.
+ *
+ * Arguments:
+ * - delta_time: How long has passed between ticks.
+ */
+/obj/item/tank/proc/handle_tolerances(delta_time)
 	if(!air_contents)
-		return 0
+		return FALSE
 
 	var/pressure = air_contents.return_pressure()
 	var/temperature = air_contents.return_temperature()
+	if(temperature >= TANK_MELT_TEMPERATURE)
+		var/temperature_damage_ratio = (temperature - TANK_MELT_TEMPERATURE) / temperature
+		take_damage(max_integrity * temperature_damage_ratio * delta_time, BURN, FIRE, FALSE, NONE)
+		if(QDELETED(src))
+			return TRUE
+
+	if(pressure >= TANK_LEAK_PRESSURE)
+		var/pressure_damage_ratio = (pressure - TANK_LEAK_PRESSURE) / (TANK_RUPTURE_PRESSURE - TANK_LEAK_PRESSURE)
+		take_damage(max_integrity * pressure_damage_ratio * delta_time, BRUTE, BOMB, FALSE, NONE)
+	return TRUE
+
+/// Handles the tank springing a leak.
+/obj/item/tank/obj_break(damage_flag)
+	if(!leaking)
+		leaking = TRUE
```

Comment:
```suggestion
	leaking = TRUE
```

If it's already TRUE, then it just won't have any effect anyway.

Reply:
Done.